catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -150,6 +151,11 @@
   private final int commitNrtCommitWithinMs =
       Math.max(NumberUtils.toInt(accessProperty(SOLR_COMMIT_NRT_COMMITWITHINMS, "1000")), 0);
 
+  private static final String SOLR_QUERY_TIMEALLOWEDMS = "solr.query.timeAllowed";
+
+  private final int queryTimeAllowedMs =
+      Math.max(NumberUtils.toInt(accessProperty(SOLR_QUERY_TIMEALLOWEDMS, "0")), 0);
```

Comment:
what would 0 do? would it not perform the search at all or omit this property? should it be default to something else?

Reply:
Before setting the time allowed on the Solr query, I check if `queryTimeAllowedMs` is greater than 0.  So this currently disables it.  I was having a hard time coming up with a safe default for everyone's production use case.

I would like to have a sensible default other than disabling it but I did not want to hold up getting this in while trying to figure out what that value should be.  Ideally it would be something like one second but that assumes everyone will be able to scale their clusters or fix their schemas (and reindex) quick enough to avoid going over that value.  I think it would be safe for the majority of production use cases but I know it is not safe for all of them.  But a value safe for all of them might be so high that it is similar to disabling it.

I am also not sure how many production use cases might have a small outlier of users that really need to do very complex queries and get the full results back.  

There is also no feedback from current endpoints to know if you are seeing the full results or a partial result.